File: salonmanager/backend/app/Models/Booking.php
<?php

namespace App\Models;

use App\Support\Tenancy\SalonOwned;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Booking extends Model
{
    use SalonOwned;

    protected $fillable = [
        'salon_id',
        'customer_id',
        'stylist_id',
        'start_at',
        'end_at',
        'status',
        'notes',
    ];

    protected $casts = [
        'start_at' => 'datetime',
        'end_at' => 'datetime',
    ];

    public const STATUS_PENDING = 'pending';
    public const STATUS_CONFIRMED = 'confirmed';
    public const STATUS_DECLINED = 'declined';
    public const STATUS_CANCELED = 'canceled';

    public function salon(): BelongsTo
    {
        return $this->belongsTo(Salon::class);
    }

    public function customer(): BelongsTo
    {
        return $this->belongsTo(User::class, 'customer_id');
    }

    public function stylist(): BelongsTo
    {
        return $this->belongsTo(Stylist::class);
    }

    public function services(): BelongsToMany
    {
        return $this->belongsToMany(Service::class, 'booking_services')
            ->withPivot('duration', 'price', 'stylist_id');
    }

    public function media(): HasMany
    {
        return $this->hasMany(BookingMedia::class);
    }
}

File: salonmanager/frontend/lib/features/booking/models/booking.dart
import 'service.dart';
import 'stylist.dart';

class Booking {
  final int id;
  final int salonId;
  final int customerId;
  final int? stylistId;
  final DateTime startAt;
  final DateTime endAt;
  final String status;
  final String? notes;
  final List<Service> services;
  final Stylist? stylist;
  final List<String> imagePaths;

  const Booking({
    required this.id,
    required this.salonId,
    required this.customerId,
    this.stylistId,
    required this.startAt,
    required this.endAt,
    required this.status,
    this.notes,
    this.services = const [],
    this.stylist,
    this.imagePaths = const [],
  });

  factory Booking.fromJson(Map<String, dynamic> json) {
    return Booking(
      id: json['id'] as int,
      salonId: json['salon_id'] as int,
      customerId: json['customer_id'] as int,
      stylistId: json['stylist_id'] as int?,
      startAt: DateTime.parse(json['start_at'] as String),
      endAt: DateTime.parse(json['end_at'] as String),
      status: json['status'] as String,
      notes: json['notes'] as String?,
      services: json['services'] != null
          ? (json['services'] as List)
              .map((s) => Service.fromJson(s as Map<String, dynamic>))
              .toList()
          : [],
      stylist: json['stylist'] != null
          ? Stylist.fromJson(json['stylist'] as Map<String, dynamic>)
          : null,
      imagePaths: json['media'] != null
          ? (json['media'] as List)
              .map((m) => m['path'] as String)
              .toList()
          : [],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'salon_id': salonId,
      'customer_id': customerId,
      'stylist_id': stylistId,
      'start_at': startAt.toIso8601String(),
      'end_at': endAt.toIso8601String(),
      'status': status,
      'notes': notes,
      'services': services.map((s) => s.toJson()).toList(),
      'stylist': stylist?.toJson(),
      'image_paths': imagePaths,
    };
  }

  bool get isPending => status == 'pending';
  bool get isConfirmed => status == 'confirmed';
  bool get isDeclined => status == 'declined';
  bool get isCanceled => status == 'canceled';
}